controllers/topics/checkName.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    let response = {
+      success: false,
+      message: 'A community with this name already exists',
+      available: false
+    };
+
+    if (topics) {
+      response = {
+        success: true,
+        message: 'Community name available',
+        available: true
+      };
+    }
+
+    return res.status(200).json(response);
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in your code seems to be reversed. If a topic with the given name is found, it means that the name is not available. However, your code sets `available` to `true` and sends a message saying "Community name available" when a topic is found. This should be corrected.

```diff


<!-- This is an auto-generated comment by OSS CodeRabbit -->